modules/core-components/src/main/java/org/mule/runtime/core/internal/routing/ProcessorRoute.java

```diff
@@ -69,11 +73,18 @@ public void setMessagingExceptionHandler(FlowExceptionHandler messagingException
     this.messagingExceptionHandler = messagingExceptionHandler;
   }
 
+  public void setSpanCustomizationInfo(SpanCustomizationInfo spanCustomizationInfo) {
+    this.spanCustomizationInfo = spanCustomizationInfo;
+  }
+
   @Override
   public void initialise() throws InitialisationException {
     if (processor instanceof MessagingExceptionHandlerAware) {
       ((MessagingExceptionHandlerAware) processor).setMessagingExceptionHandler(messagingExceptionHandler);
     }
+    if (processor instanceof SpanCustomizationInfoAware) {
```

Comment:
Not a fan of having to create an interface to add the SpanCustomizationInfo to the processor, but couldn't find another way, since at that point `processor` is an instance of `LazyProcessorChainBuilder` which is a private class, child of `AbstractMessageProcessorChain` which is a package private class so that was the only way I could think of that would allow me to call `setSpanCustomizationInfo`. I accept other suggestions if they come to mind

Reply:
Neither do I. I see why it is needed.